### ISSUE TITLE
How to fix patch compatibility on Windows XP

### ISSUE THREAD

**Issue by bionicle159** [CookiePLMonster/SilentPatchBully#55]
Hi,
Thought I'd write my findings with getting this patch to run on XP for people who run retro machines. 
Since the included 'dinput8.dll' isn't compatible with XP, I swapped it around with the included one in XPs System32 and got it to run. Still acts sluggish on load-up, but once everything is loaded it seems to run fine with V-Sync turned off in-game (but forced on in graphics drivers).
If anyone can double-check my findings and make sure it checks out then it would be good to include both options when downloading the patch.

**Comment by CookiePLMonster** [comment]
If you've replaced dinput8.dll (from Ultimate ASI Loader) with the OS version, then SP is not loaded and thus you might as well have removed all the mod files.

The easiest way to get to it would be to fork UAL and strip it out of anything which is not compatible with XP, and recompile SPBully with a _xp toolset since other than that it should "just work".

I usually keep XP compatibility in existing projects, but don't target it with anything new - especially since even if SPBully itself was compatible with XP, its loader is not.

**Comment by bionicle159** [comment]
I'm a casual user so I wouldn't know where to begin with recompiling the program. Even if it might be a placebo, it did seem like the toolset still worked with the changed 'dll', the only issue it still has is the caching taking a few seconds, but otherwise it's fine with the framecap turned off - since adding the patch, entering the game is instantaneous and doesn't have the frame drops it did before.

**Comment by CookiePLMonster** [comment]
It is placebo at its finest. Like I said, by replacing this DLL you made SPBully not load at all. You were playtesting a stock game.

**Comment by bionicle159** [comment]
If that's the case and if modifying is straightforward, why don't you create a port of the patch for XP users - if it's as simple as you said it was then it shouldn't be much effort for to implement.

**Comment by CookiePLMonster** [comment]
I had a look and it wouldn't be as straightforward after all, for several reasons:
- I don't want to maintain UAL's fork.
- SPBully **does** use WinAPI calls introduced in Vista and they are imperative for one of the fixes to function correctly. I don't know if fallback is possible.
- XP is unsupported for 6 years by now and the % of people still using it is so low so I really doubt it's worth spending time on.

**Comment by bionicle159** [comment]
It's Github, if anyone who does have the time and knowledge to fix it for XP, eventually they'll come forward with the solution.

**Comment by CookiePLMonster** [comment]
> It's Github, if anyone who does have the time and knowledge to fix it for XP, eventually they'll come forward with the solution.

You'd be surprised how very few people care about XP, out of all people I know who are also releasing mods/plugins I'm one of the last not to drop XP completely from the existing projects.

This patch has been around for over 2 years now and it's been open source since day one, and nobody wanted to do anything regarding XP support. I'm not saying it'll never happen, but the odds of it are very, very low.

**Comment by bionicle159** [comment]
No-one thought new MIDI sound cards and 68K CPUs  would be made decades after ISA and the Amiga died off, but here we are. All it takes is for people to get nostalgic and in need of an option. I doubt anyone will benefit from this thread but at least it's out there to be picked up if anyone can help.
